src/configuration.js

```diff
@@ -365,6 +362,10 @@ Configuration.prototype._checkLocalProjectId = function(cb) {
     } else {
       cb(new Error('config.projectId must be a string or number'), null);
     }
+  } else if (isString(env.GCLOUD_PROJECT) && !isEmpty(env.GCLOUD_PROJECT)) {
```

Comment:
I will comment it, but its cheaper to switch on type detection first and then do a more abstract analysis of 'emptiness' afterwards which itself includes a type switch.